### PR TITLE
Meta: integrate esmeta type checker into CI

### PR DESCRIPTION
**Pull request by michaelficarra**
Currently has 72 false positives. Waiting on:

- [x] esmeta support for accepting an exclusions file to ignore remaining false positives
- [x] #2923 
- [ ] #2924 
- [x] #2925 
- [ ] possibly wait for more esmeta typechecker bugfixes to reduce the number of ignored false positives

### CODE REVIEW

**Review by ljharb — COMMENTED** [review]
This can't be made to run locally with `npm install && npm run something`?

File: .github/workflows/esmeta-typecheck.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 17
+      - name: download esmeta
+        run: 'git clone --depth 1 --branch v0.1.0-rc3 https://github.com/es-meta/esmeta.git "${ESMETA_HOME}"'
```
[the remark this answers is not in the document]
Comment (michaelficarra) [reply .github/workflows/esmeta-typecheck.yml:22]: Is that better though? This says exactly what's going on.

**Comment by bakkot** [comment]
For exclusions support, is there an error if something excluded does _not_ have an error? We'll never remember to remove stuff from it otherwise.

**Comment by jhnaldo** [comment]
Hi, @bakkot. Do you mean that you want ESMeta to throw an error when excluded algorithms do not have any type mismatches?

**Comment by jhnaldo** [comment]
@bakkot Thank you for the kind explanation! I supported the feature you mentioned to detect unused names listed in the "excluded" file in the [ESMeta v0.1.0-rc5 version](https://github.com/es-meta/esmeta/releases/tag/v0.1.0-rc5). And, I attached the tight `esmeta-ignore.json` file as follows:
```json
[
  "AddEntriesFromIterable",
  "AsyncGeneratorBody[0,0].EvaluateAsyncGeneratorBody",
  "BoundFunctionCreate",
  "Catch[1,0].CatchClauseEvaluation",
  "ClassElement[0,0].ClassElementEvaluation",
  "ClassElement[1,0].ClassElementEvaluation",
  "ClassStaticBlockBody[0,0].EvaluateClassStaticBlockBody",
  "CreateArrayIterator",
  "CreateBuiltinFunction",
  "CreateDynamicFunction",
  "CreateMappedArgumentsObject",
  "DeclarativeEnvironmentRecord.InitializeBinding",
  "DeclarativeEnvironmentRecord.SetMutableBinding",
  "FlattenIntoArray",
  "FunctionBody[0,0].EvaluateFunctionBody",
  "GeneratorBody[0,0].EvaluateGeneratorBody",
  "GeneratorResumeAbrupt",
  "GeneratorValidate",
  "GetGlobalObject",
  "GetMethod",
  "GetNewTarget",
  "GetPromiseResolve",
  "GetPrototypeFromConstructor",
  "GetThisValue",
  "INTRINSICS.DefaultTimeZone",
  "InnerModuleEvaluation",
  "IntegerIndexedObjectCreate",
  "LabelledItem[1,0].LabelledEvaluation",
  "LengthOfArrayLike",
  "MethodDefinition[0,0].DefineMethod",
  "ModuleNamespaceCreate",
  "NewPromiseReactionJob",
  "NormalCompletion",
  "OrdinaryGet",
  "PrivateGet",
  "ProxyCreate",
  "RegExpInitialize",
  "SerializeJSONObject",
  "SerializeJSONProperty",
  "SetDefaultGlobalBindings",
  "SetViewValue",
  "SortIndexedProperties",
  "SourceTextModuleRecord.ResolveExport",
  "SpeciesConstructor",
  "Statement[0,0].LabelledEvaluation",
  "Statement[1,0].LabelledEvaluation",
  "Statement[2,0].LabelledEvaluation",
  "StringCreate",
  "StringPad",
  "TemplateString",
  "ToBigInt64",
  "ToBigUint64",
  "ToUint32",
  "TrimString",
  "WeakRefDeref"
]
```

**Review by bakkot — APPROVED** [review]

**Pull request merged**